File: LeetCode/Easy/1491-average-salary-excluding-the-minimum-and-maximum-salary/1491-average-salary-excluding-the-minimum-and-maximum-salary.java
class Solution {
    public double average(int[] salary) {
        double ans = 0D;

        int max = Integer.MIN_VALUE;
        int min = Integer.MAX_VALUE;

        for(int i = 0; i < salary.length; ++i) {
            ans += salary[i];
            max = Math.max(max, salary[i]);
            min = Math.min(min, salary[i]);
        }

        if(max == min) {
            ans -= max;
            return ans / (salary.length - 1);
        } else {
            ans -= max;
            ans -= min;
            return ans / (salary.length - 2);
        }
    }
}

File: LeetCode/Easy/1491-average-salary-excluding-the-minimum-and-maximum-salary/1491-average-salary-excluding-the-minimum-and-maximum-salary.kt
class Solution {
    fun average(salary: IntArray): Double {
        var ans: Double = 0.0

        var max: Int = Int.MIN_VALUE
        var min: Int = Int.MAX_VALUE

        for(s in salary) {
            ans += s;
            max = Math.max(max, s)
            min = Math.min(min, s)
        }


        if(max == min) {
            ans -= max
            return ans / (salary.size - 1)
        } else {
            ans -= max
            ans -= min
            return ans / (salary.size - 2)
        }
        
    }
}